cassiopeia/data/CassiopeiaTree.py

```diff
@@ -1872,6 +1872,105 @@ def get_distances(
             for _node in (self.leaves if leaves_only else self.nodes)
         }
 
+    def scale_to_unit_length(self) -> None:
+        """
+        Scales the tree to have unit length.
+
+        The longest path from root to leaf will have length 1 after the
+        scaling.
+
+        Raises:
+            CassiopeiaTreeError if the tree has not been initialized.
+        """
+        self.__check_network_initialized()
+
+        current_times = self.get_times().values()
+        max_time = max(current_times)
+        min_time = min(current_times)
+        new_times = {}
+        for node in self.nodes:
+            new_times[node] = self.get_time(node) / (max_time - min_time)
+        self.set_times(new_times)
+
+    def get_depth(self) -> float:
+        """
+        Depth of the tree (in terms of time).
+
+        The depth of the tree (in terms of time) is defined as the greatest
+        time distance of any leaf of the tree from the root.
+
+        Returns:
+            Depth of the tree.
+
+        Raises:
+            CassiopeiaTreeError if the tree has not been initialized.
+        """
+        self.__check_network_initialized()
+
+        times = self.get_times().values()
+        return max(times) - min(times)
+
+    def get_mutated_characters_along_edge(
+        self, parent: str, child: str
+    ) -> List[int]:
+        """
+        List of mutated characters along edge.
+
+        A character is considered to mutate if it goes from the zero state
+        to a non-zero state that is also not a missing state. In particular,
+        if a character goes missing, it is NOT considered as mutated by this
+        method.
+
+        Raises:
+            CassiopeiaTreeError if the edge does not exist or if the tree is
+                not initialized.
+        """
+        self.__check_network_initialized()
+
+        if child not in self.children(parent):
+            raise CassiopeiaTreeError("Edge does not exist.")
+
+        parent_states = self.get_character_states(parent)
+        child_states = self.get_character_states(child)
+        res = []
+        for i, (parent_state, child_state) in enumerate(
+            zip(parent_states, child_states)
+        ):
+            if (
+                parent_state == 0
+                and child_state != 0
+                and child_state != self.missing_state_indicator
```

Comment:
Why not be a bit more general here?

I think the following check would have the same effect and yet be general to reversible models:
```
if parent_state != child_state
and child_state != self.missing_state_indicator
and parent_state != self.missing_state_indicator:
```

Reply:
Yeah I like this!

By the way, note that `CassiopeiaTree` already implements a method `get_mutations_along_edge` which assumes that missing characters are mutated characters (so it doesn't have the behavior I need). The returned object is also a `List[Tuple[int, int]]` instead of a `List[int]`.

I don't think that `get_mutations_along_edge` is used anywhere in the codebase. We can try to unify both methods, of keep both, or even remove `get_mutations_along_edge`. What do you think?

Reply:
Good catch! I had forgotten that we had implemented this function. As it is, I think you're right - we're not using `get_mutations_along_edge` explicitly in the codebase but from my previous work analyzing these trees, I actually prefer returning `List[Tuple[int, int]]`. Sometimes you just want the mutated characters (like you are) and sometimes you actually want the mutated states (e.g., for computing likelihoods).

So, I would prefer to remove your function `get_mutated_characters_along_edge` and adapt the existing `get_mutations_along_edge` to your specifications by adding extra arguments.

Reply:
Done